ops/_main.py

```diff
@@ -364,31 +364,35 @@ class _Manager:
       - the event that Juju is emitting on us
       - the charm instance (user-facing)
     - emit: core user-facing lifecycle step. Consists of:
-      - re-emit any deferred events found in the storage
       - emit the Juju event to the charm
         - emit any custom events emitted by the charm during this phase
       - emit  the ``collect-status`` events
     - commit: responsible for:
       - store any events deferred throughout this execution
       - graceful teardown of the storage
+
+    The above steps are first run for any deferred notices found in the storage
+    (all three steps for each notice, except for emitting Lifecycle events), and
+    then run a final time for the Juju event that triggered this execution.
     """
 
     def __init__(
         self,
         charm_class: Type['_charm.CharmBase'],
-        model_backend: Optional[_model._ModelBackend] = None,
+        juju_context: _JujuContext,
         use_juju_for_storage: Optional[bool] = None,
         charm_state_path: str = CHARM_STATE_FILE,
-        juju_context: Optional[_JujuContext] = None,
     ):
-        if juju_context is None:
-            juju_context = _JujuContext.from_dict(os.environ)
```

Comment:
Ops and Scenario are the only ones creating `_Manager` objects, and now that we're forcing alignment of versions of `ops` and `ops-scenario`, this compatibility code can be removed to simplify the class.